src/main/java/com/bootcamp/demo/dao/FirestoreDaoImpl.java

```diff
@@ -156,4 +176,15 @@ public Card setPreferredCard(String cardNumber) {
         }
         return objects;
     }
+
+    @Override
+    public Transaction addTransaction(Transaction transaction) {
+        Map<String, Object> transactionsData = new HashMap<>();
+        transactionsData.put("cardNumber", transaction.getCardNumber());
+        transactionsData.put("amount", transaction.getAmount());
+        transactionsData.put("timestamp", transaction.getTimestamp());
```

Comment:
You could save the timestamp as epoch time `.toEpochSecond(ZoneOffset.UTC)`. The epoch time is an universal time format → https://www.epochconverter.com/